### ISSUE TITLE
cy.route() - "No request ever occurred"

### ISSUE THREAD

**Issue by dotherightthing** [cypress-io/cypress#2507]
I have a modal which is populated via Ajax. I am trying to watch a route so that I can run my tests only once the Ajax response (the modal HTML) has come in. I have verified that the 3rd party Modaal library is making the `$.ajax` request (`console.log`), that the there is an XHR request and response (Dev Tools Network tab), and that the request is successful (the HTML renders inside the modal). I've tried using both a local and remote test page, but Cypress consistently reports that no request was ever made. 

I posted this on Gitter and @bahmutov asked me to log this as an issue. 

### Current behavior:

1. Create a route
2. Click link to launch the modal (`href` is Ajax target)
3. Modaal script makes an Ajax request
4. Cypress announces that a click was made and it is waiting for the response
5. Cypress times out, with the error "No request ever occurred."

### Desired behavior:

1. Create a route
2. Click link to launch the modal (`href` is Ajax target)
3. Cypress announces that a click was made
4. Modaal script makes an Ajax request
5. Cypress announces that it is waiting for the response
6. Cypress passes, acknowledging that the HTML response was received

### Steps to reproduce:

1. Clone https://github.com/dotherightthing/cypress-test-tiny-ajax-html
2. Run cypress/integration/ajax-html/spec.js

### Versions

Cypress 3.1.0
Browser: Chrome 69.0.3497.100 (Official Build) (64-bit)
OS: MacOS Sierra 10.12.6

**Comment by Saibamen** [comment]
Confirmed with Cypress 3.3.0, Chrome 74 and Electron 61.
Windows 10.0.17763.504

**Comment by jennifer-shehane** [comment]
By default, some resources are whitelisted from ever being logged or stubbed - this is mostly to prevent noise in the Command Log and to prevent confusion between urls matching XHR requests that are non-static resources versus static resources.

You can, however, [change this whitelisting behavior](https://on.cypress.io/server#Change-the-default-whitelisting) so that your html resources will be waited upon properly. 

This is the final code I came up with:

```js
describe('Modaal', function () {
  it('User can launch the modal', function () {    
    // load local web page
    cy.visit('cypress/integration/ajax-html/page.html');

    cy.server({
      whitelist: (xhr) => {
        // this function receives the xhr object in question and
        // will whitelist if it's a GET that appears to be a static js or css
        return xhr.method === 'GET' && /\.(jsx?|css)(\?.*)?$/.test(xhr.url)
      }
    });
    cy.route('**/cypress/integration/ajax-html/modal.html').as('modalAjaxHtml');

    // check that the launch link exists
    cy.get('a.modaal-ajax').click();

    // wait for the Ajax response
    cy.wait('@modalAjaxHtml').then(function(xhr){
      console.log('Cypress: Ajax response', xhr);
    });
  });
});
```

**Comment by Saibamen** [comment]
Doesn't work for me.

```
describe('Section1Overview', function () {
    before(() => {
        cy.login()
        cy.server({
            whitelist: (xhr) => {
                return xhr.method === 'GET' || xhr.method === 'POST'
            }
        });
        cy.route('GET', '*/Assessment/*', []).as('loadQuestionnaire');
        cy.visit(Cypress.env('employee_form'))
        cy.wait('@loadQuestionnaire')
    })

    require('./Section1/Page1.js')
})
```

I fetch GET for `https://xxx.yyy.info/api/Assessment/loadEmployeeQuestionnaire` after visiting my `employee_form`

**Comment by jennifer-shehane** [comment]
@Saibamen I am not sure what you are experiencing is the same issue. The initial issue was involved with a `.html` resource loading. 

The glob pattern you've provide `*/Assessment/*` does not match the url given. You would need to use `**/Assessment/*`. Please check your globs with minimatch or an online glob tester.

**Comment by Saibamen** [comment]
Still doesn't work...

http://www.globtester.com/#p=eJzT0tJ3LC5OLS7OTc0r0dcCACYcBQM%3D&r=eJztksFOhDAQhl9lwsHsJl26iLtGYEmM0ZN7UJ%2BgwMA2lg62xYBPbxETYzx5lHjopZ3O%2FP%2F%2FTVbJVyiVsPYQtMKVpyDPbCc0WDcqPAQlKTIJmKZYxXsGF5E%2F%2B3UKNWm3qUUr1ZjA092RNG0esemVMAxuSFvyPRmcvfTk0ntZoBFOkoapcL5kcEStaKrujUT%2FrfVvfnaJn%2B2tfMMEoii86oYUClE%2BN4Z6XW2%2BRInV%2BSWD2KuKdwy24Xa3ToP85FxnE86HYQjHcQylromLTvJra9HaFrXjikR123aKRsSHHu2kTgtpMOOT%2FzwrTJ5xn06eLTGi36bx5w3PPGczP7AugejHgot%2FpAtC6jy1bwbfASgz4e0%3D&

**Comment by dotherightthing** [comment]
Thanks for the workaround. I'm currently travelling but will try this out when I return home in August.

**Comment by Saibamen** [comment]
My issue is still existing in Cypress 3.3.1

**Comment by jennifer-shehane** [comment]
@Saibamen The globtester is unfortunately quite flawed - it doesn't highlight correctly anymore if you paste text in, you have to type it directly. Please open a new issue exactly describing what you are experiencing. I do not think it is related to this .html issue at all.
